internal/fs/vss_windows.go

```diff
@@ -0,0 +1,986 @@
+		// snapshot because we need to consider this when backing up files
+		mountPointInfo[mountPoint] = MountPoint{isSnapshotted: false}
+
+		if isSupported, err := iVssBackupComponents.IsVolumeSupported(mountPoint); err != nil {
+			continue
+		} else if !isSupported {
+			continue
+		}
+
+		var mountPointSnapshotSetID ole.GUID
+		err := iVssBackupComponents.AddToSnapshotSet(mountPoint, &mountPointSnapshotSetID)
+		if err != nil {
+			iVssBackupComponents.Release()
+			return VssSnapshot{}, err
+		}
+
+		mountPointInfo[mountPoint] = MountPoint{isSnapshotted: true,
+			snapshotSetID: mountPointSnapshotSetID}
+	}
+
+	err = callAsyncFunctionAndWait(iVssBackupComponents.PrepareForBackup, "PrepareForBackup",
+		timeoutInMillis)
+	if err != nil {
+		// After calling PrepareForBackup one needs to call AbortBackup() before releasing the VSS
+		// instance for proper cleanup.
+		// It is not neccessary to call BackupComplete before releasing the VSS instance afterwards.
+		iVssBackupComponents.AbortBackup()
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	err = callAsyncFunctionAndWait(iVssBackupComponents.DoSnapshotSet, "DoSnapshotSet",
+		timeoutInMillis)
+	if err != nil {
+		iVssBackupComponents.AbortBackup()
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	var snapshotProperties VssSnapshotProperties
+	err = iVssBackupComponents.GetSnapshotProperties(snapshotSetID, &snapshotProperties)
+	if err != nil {
+		iVssBackupComponents.AbortBackup()
+		iVssBackupComponents.Release()
+		return VssSnapshot{}, err
+	}
+
+	for mountPoint, info := range mountPointInfo {
+		err := iVssBackupComponents.GetSnapshotProperties(info.snapshotSetID,
+			&info.snapshotProperties)
+		if err != nil {
+			msgError(mountPoint, errors.Errorf(
+				"VSS error: GetSnapshotProperties() for mount point %s returned error: ",
+				mountPoint, err))
+			info.isSnapshotted = false
+		} else {
+			info.snapshotDeviceObject = info.snapshotProperties.GetSnapshotDeviceObject()
+		}
+
+		mountPointInfo[mountPoint] = info
+	}
+
+	return VssSnapshot{iVssBackupComponents, snapshotSetID, snapshotProperties,
+		snapshotProperties.GetSnapshotDeviceObject(), mountPointInfo, timeoutInMillis}, nil
+}
+
+// Delete deletes the created snapshot.
+func (p *VssSnapshot) Delete() error {
+	var err error
+	if err = vssFreeSnapshotProperties(&p.snapshotProperties); err != nil {
+		return err
+	}
+
+	for _, mountPoint := range p.mountPointInfo {
+		if mountPoint.isSnapshotted {
+			if err = vssFreeSnapshotProperties(&mountPoint.snapshotProperties); err != nil {
+				return err
+			}
+		}
+	}
+
+	if p.iVssBackupComponents != nil {
+		defer p.iVssBackupComponents.Release()
+
+		err = callAsyncFunctionAndWait(p.iVssBackupComponents.BackupComplete, "BackupComplete",
+			p.timeoutInMillis)
+
+		if _, _, err := p.iVssBackupComponents.DeleteSnapshots(p.snapshotID); err != nil {
+			err = newVssTextError(fmt.Sprintf("Failed to delete snapshot:  %s", err.Error()))
```

Comment:
Here you write to 'if' scoped `err`, not `err` declared in the beginning of function.
Regardless, this code looks strange.
If you fixing it: if `BackupComplete` failed it's `err` value will be lost if `DeleteSnapshots` failed too.
If you not: if `DeleteSnapshots` failed it's `err` value will be lost.

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/97322db98cdc31a9aba2c358e59809418cdf68be